components/automate-cli/cmd/chef-automate/cmdUtils.go

```diff
@@ -73,132 +80,131 @@ func (c *remoteCmdExecutor) Execute() error {
 		if err != nil {
-			return err
+			return map[string][]CmdResult{}, err
 		}
 
-		c.executeCmdOnGivenNodes(c.NodeMap.Postgresql.CmdInputs, nodeIps, remoteService, timestamp, writer)
+		output := c.executeCmdOnGivenNodes(c.NodeMap.Postgresql.CmdInputs, nodeIps, remoteService, timestamp, writer)
+		return output, nil
 	case c.NodeMap.Opensearch.CmdInputs.NodeType:
 		const remoteService string = CONST_OPENSEARCH
 		nodeIps, err := preCmdExecCheck(c.NodeMap.Opensearch, c.SshUtil, c.NodeMap.Infra, remoteService, timestamp, writer)
 		if err != nil {
-			return err
+			return map[string][]CmdResult{}, err
 		}
 
-		c.executeCmdOnGivenNodes(c.NodeMap.Opensearch.CmdInputs, nodeIps, remoteService, timestamp, writer)
+		output := c.executeCmdOnGivenNodes(c.NodeMap.Opensearch.CmdInputs, nodeIps, remoteService, timestamp, writer)
+		return output, nil
 	default:
-		return errors.New("Missing or Unsupported flag")
+		return map[string][]CmdResult{}, errors.New("Missing or Unsupported flag")
 	}
-
-	return nil
 }
 
-// executeCmdOnGivenNodes executes given command/commands on all given nodes concurrently.
-func (c *remoteCmdExecutor) executeCmdOnGivenNodes(input *CmdInputs, nodeIps []string, remoteService string, timestamp string, cliWriter *cli.Writer) {
-	resultChan := make(chan CmdResult, len(nodeIps))
+func (c *remoteCmdExecutor) executeCmdOnGivenNodes(input *CmdInputs, nodeIps []string, remoteService string, timestamp string, cliWriter *cli.Writer) map[string][]CmdResult {
+	command := make(map[string]string)
+	ouputJsonResultMapMutex := sync.RWMutex{}
 	originalSSHConfig := c.SshUtil.getSSHConfig()
 	inputFiles := input.InputFiles
 	outputFiles := input.Outputfiles
-	command := input.Cmd
+	if len(input.MutipleCmdWithArgs) == 0 {
+		command["Script"] = input.Cmd
+	} else {
+		command = input.MutipleCmdWithArgs
+	}
 	timeout := input.WaitTimeout
-
 	inputFileToOutputFileMap := map[string]string{}
 	for _, file := range inputFiles {
 		destinationFile := remoteService + "_" + timestamp + "_" + file
 		inputFileToOutputFileMap[file] = destinationFile
 	}
-
-	for _, hostIP := range nodeIps {
-		newSSHConfig := &SSHConfig{
-			sshUser:    originalSSHConfig.sshUser,
-			sshPort:    originalSSHConfig.sshPort,
-			sshKeyFile: originalSSHConfig.sshKeyFile,
-			hostIP:     hostIP,
-			timeout:    timeout,
-		}
-		sshUtil := NewSSHUtil(newSSHConfig)
-
-		printConnectionMessage(remoteService, hostIP, cliWriter)
-
-		go c.executeCmdOnNode(command, inputFileToOutputFileMap, outputFiles, remoteService, input.ErrorCheckEnableInOutput, sshUtil, resultChan)
+	ouputJsonResult := make(map[string][]CmdResult)
+	var wg sync.WaitGroup
+	for scriptName, cmdScript := range command {
+
+		wg.Add(1)
+		go func(originalSSHConfig *SSHConfig, timeout int, cmdScript string, scriptName string, inputFileToOutputFileMap map[string]string, outputFiles []string, remoteService string, errorCheckEnableInOutput bool) {
+			defer wg.Done()
+			for _, hostIP := range nodeIps {
```

Comment:
run the go routine inside this loop over nodeIps so that the the code runs concurrently for all nodeIps*commands

Reply:
its running concurrently, if i move loop nodeIP outside i am getting error